File: Old C# Codes/Dash.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DebateTournamentTabSystem.BLL
{
    public class Dash
    {
        public static Tournament CreateTournament()
        {
            Console.Clear();
            Console.Write("Enter tournament name: ");
            string name = Console.ReadLine();

            Console.Write("Enter tournament year: ");
            string year = Console.ReadLine();

            Console.Write("Enter club name: ");
            string club = Console.ReadLine();

            // Simplify the 'new' expression in the CreateTournament method
            Tournament newTournament = new Tournament(club, name, year);
            ExcelExporter.SaveTournamentToExcel(newTournament);
            return newTournament;

            Console.WriteLine("Tournament created successfully. Press any key to return to the main menu!");
            Console.ReadKey();
        }
        public static void RunTournamentMenu(Tournament tournament)
        {
            while (true)
            {
                Console.Clear();
                Console.WriteLine($"Running Tournament: {tournament.tournamentName} ({tournament.tournamentYear})");
                Console.WriteLine($"Current Segment: {tournament.currentSegment}\n");

                Console.WriteLine("1. Add a team");
                Console.WriteLine("2. Remove a team");
                Console.WriteLine("3. View all teams");
                Console.WriteLine("4. Show Matchups for Current Segment");
                Console.WriteLine("5. Enter Match Results");
                Console.WriteLine("6. Export to Excel");
                Console.WriteLine("7. Advance Segment");
                Console.WriteLine("8. Back to main menu");
                Console.WriteLine("9. Get Tops");


                Console.Write("Enter your choice: ");
                string choice = Console.ReadLine();
                switch (choice)
                {
                    case "1":
                        Tournament.AddTeam(tournament);
                        break;
                    case "2":
                        Tournament.RemoveTeam(tournament);
                        break;
                    case "3":
                        Tournament.ViewTeams(tournament);
                        break;
                    case "4":
                        MatchMashup.ShowMatchups(tournament);
                        break;
                    case "5":
                        ScoreManager.EnterMatchResults(tournament);
                        break;
                    case "6":
                        ExcelExporter.SaveTournamentToExcel(tournament);
                        Console.WriteLine("Press any key to continue...");
                        Console.ReadKey();
                        return;
                    case "7":
                        ScoreManager.AdvanceSegment(tournament);
                        break;
                    case "8":
                        return;
                    case "9":
                        ScoreManager.ShowTopPerformers(tournament);
                        break;

                    default:
                        Console.WriteLine("Invalid choice. Press any key to try again.");
                        Console.ReadKey();
                        break;
                }
            }
        }
        public static bool ConfirmExit()
        {
            Console.Write("Are you sure you want to exit? (y/n): ");
            string input = Console.ReadLine().ToLower();
            return input == "y" || input == "yes";
        }
    }
}

File: lib/controllers/dash.dart
// import 'dart:io';
// import '../models/tournament.dart';
// import '../models/tournament_segment.dart';
// import '../services/data_manager.dart';
// import '../services/match_mashup.dart';
// import '../services/score_manager.dart';

// class Dash {
//   static Future<Tournament> createTournament({
//     Future<String> Function(String)? inputCallback,
//     Function(String)? onMessage,
//   }) async {
//     String name, year, club;

//     if (inputCallback != null) {
//       name = await inputCallback("Enter tournament name:");
//       year = await inputCallback("Enter tournament year:");
//       club = await inputCallback("Enter club name:");
//     } else {
//       print("Enter tournament name:");
//       name = stdin.readLineSync() ?? '';
//       print("Enter tournament year:");
//       year = stdin.readLineSync() ?? '';
//       print("Enter club name:");
//       club = stdin.readLineSync() ?? '';
//     }

//     Tournament newTournament = Tournament(
//       clubName: club,
//       tournamentName: name,
//       tournamentYear: year,
//     );

//     await DataManager.saveTournamentToJson(newTournament);

//     if (onMessage != null) {
//       onMessage("Tournament created successfully!");
//     } else {
//       print(
//           "Tournament created successfully. Press any key to return to the main menu!");
//     }

//     return newTournament;
//   }

//   static Future<void> runTournamentMenu(
//     Tournament tournament, {
//     Function(String, List<String>)? onShowMenu,
//     Future<String> Function(String)? inputCallback,
//     Function(String)? onMessage,
//     Function()? onReturn,
//   }) async {
//     while (true) {
//       List<String> menuOptions = [
//         "Add a team",
//         "Remove a team",
//         "View all teams",
//         "Show Matchups for Current Segment",
//         "Enter Match Results",
//         "Export Data",
//         "Advance Segment",
//         "Back to main menu",
//         "Get Tops",
//       ];

//       String menuTitle =
//           "Running Tournament: ${tournament.tournamentName} (${tournament.tournamentYear})\n"
//           "Current Segment: ${tournament.currentSegment.displayName}";

//       String choice;
//       if (onShowMenu != null) {
//         onShowMenu(menuTitle, menuOptions);
//         choice = await inputCallback?.call("Enter your choice:") ?? '';
//       } else {
//         print(menuTitle);
//         for (int i = 0; i < menuOptions.length; i++) {
//           print("${i + 1}. ${menuOptions[i]}");
//         }
//         print("Enter your choice:");
//         choice = stdin.readLineSync() ?? '';
//       }

//       switch (choice) {
//         case "1":
//           await Tournament.addTeam(
//             tournament,
//             inputCallback: inputCallback,
//             onSuccess: onMessage,
//             onError: onMessage,
//           );
//           await DataManager.saveTournamentToJson(tournament);
//           break;

//         case "2":
//           await Tournament.removeTeam(
//             tournament,
//             inputCallback: inputCallback,
//             onSuccess: onMessage,
//             onError: onMessage,
//           );
//           await DataManager.saveTournamentToJson(tournament);
//           break;

//         case "3":
//           Tournament.viewTeams(tournament);
//           break;

//         case "4":
//           MatchMashup.showMatchups(tournament, onMessage: onMessage);
//           await DataManager.saveMatchupToJson(
//               tournament.currentMatches, tournament);
//           break;

//         case "5":
//           await ScoreManager.enterMatchResults(
//             tournament,
//             inputCallback: inputCallback,
//             onMessage: onMessage,
//           );
//           await DataManager.saveTournamentToJson(tournament);
//           break;

//         case "6":
//           await DataManager.saveTournamentToJson(tournament);
//           await DataManager.exportTournamentToCsv(tournament);
//           if (onMessage != null) {
//             onMessage("Data exported successfully!");
//           } else {
//             print("Data exported successfully! Press any key to continue...");
//           }
//           break;

//         case "7":
//           await ScoreManager.advanceSegment(tournament, onMessage: onMessage);
//           await DataManager.saveTournamentToJson(tournament);
//           break;

//         case "8":
//           if (onReturn != null) {
//             onReturn();
//           }
//           return;

//         case "9":
//           ScoreManager.showTopPerformers(tournament, onMessage: onMessage);
//           var topTeams = ScoreManager.getTopTeams(
//               tournament.teamsInTheTournament, tournament);
//           var topDebaters = ScoreManager.getTopDebaters(
//               tournament.teamsInTheTournament, tournament);
//           await DataManager.saveTopTeamsToJson(topTeams, tournament);
//           await DataManager.saveTopDebatersToJson(topDebaters, tournament);
//           break;

//         default:
//           if (onMessage != null) {
//             onMessage("Invalid choice. Please try again.");
//           } else {
//             print("Invalid choice. Press any key to try again.");
//           }
//           break;
//       }
//     }
//   }

//   static Future<bool> confirmExit({
//     Future<String> Function(String)? inputCallback,
//     Future<bool> Function(String)? confirmCallback,
//   }) async {
//     if (confirmCallback != null) {
//       return await confirmCallback("Are you sure you want to exit?");
//     }

//     if (inputCallback != null) {
//       String input =
//           await inputCallback("Are you sure you want to exit? (y/n):");
//       return input.toLowerCase() == "y" || input.toLowerCase() == "yes";
//     }

//     print("Are you sure you want to exit? (y/n):");
//     String input = (stdin.readLineSync() ?? '').toLowerCase();
//     return input == "y" || input == "yes";
//   }
// }
